### PR TITLE
fix: avoid logging env in unit tests

### PR DESCRIPTION
**Pull request by johnstcn**
Seen here: https://github.com/coder/coder/actions/runs/6317011693/job/17152855571#step:5:4736

Although Github Actions does auto-censor this sort of thing, better safe than sorry.

### CODE REVIEW

File: cli/ssh_test.go [hunk 1]
```diff
@@ -368,7 +368,7 @@ func TestSSH(t *testing.T) {
 		// Ensure that SSH_AUTH_SOCK is set.
 		// Linux: /tmp/auth-agent3167016167/listener.sock
 		// macOS: /var/folders/ng/m1q0wft14hj0t3rtjxrdnzsr0000gn/T/auth-agent3245553419/listener.sock
-		pty.WriteLine("env")
+		pty.WriteLine(`grep --perl-regexp --binary-files=text 'SSH_AUTH_SOCK=[^\x00]+' /proc/self/environ`)
```
Comment (mafredri) [inline cli/ssh_test.go:371]: This won't work on macOS due to `/proc` and `--perl-regexp`. I'd suggest just doing `env | grep SSH_AUTH_SOCK=` for the best portability (note no requirement on char after `=` so as to show empty values as well.

File: pty/start_other_test.go [hunk 2]
```diff
@@ -54,7 +54,7 @@ func TestStart(t *testing.T) {
 				Height: 24,
 			},
 		}))
-		pty, ps := ptytest.Start(t, pty.Command("env"), opts)
+		pty, ps := ptytest.Start(t, pty.Command("grep", "--binary-files=text", `SSH_TTY=/dev/`, "/proc/self/environ"), opts)
```
Comment (mafredri) [inline pty/start_other_test.go:57]: (Mostly) same as above.

**Review by mtojek — APPROVED** [review]

File: cli/ssh_test.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (johnstcn) [reply cli/ssh_test.go]: I was trying to find a way that worked consistently between `pty.WriteLine` and `pty.Command`, there seem to be some subtle differences between the two.

Comment (johnstcn) [reply cli/ssh_test.go]: Updated!

**Pull request merged**